internal/pkg/manifest/svc.go

```diff
@@ -272,17 +273,49 @@ func (a *AdvancedCount) validScalingFields() []string {
 	}
 }
 
-func (a *AdvancedCount) hasScalingFieldsSet() bool {
-	switch a.workloadType {
-	case LoadBalancedWebServiceType:
-		return !a.CPU.IsEmpty() || !a.Memory.IsEmpty() || !a.Requests.IsEmpty() || !a.ResponseTime.IsEmpty()
-	case BackendServiceType:
-		return !a.CPU.IsEmpty() || !a.Memory.IsEmpty() || !a.Requests.IsEmpty() || !a.ResponseTime.IsEmpty()
-	case WorkerServiceType:
-		return !a.CPU.IsEmpty() || !a.Memory.IsEmpty() || !a.QueueScaling.IsEmpty()
-	default:
-		return !a.CPU.IsEmpty() || !a.Memory.IsEmpty() || !a.Requests.IsEmpty() || !a.ResponseTime.IsEmpty() || !a.QueueScaling.IsEmpty()
+func (a *AdvancedCount) getValidScalingFieldsSet() ([]string, error) {
```

Comment:
```suggestion
func (a *AdvancedCount) validWkldScalingFields() ([]string, error) {
```

Reply:
This change sounds less descriptive to me of what the function does. It makes it sound like it would then return the same information as `validScalingFields()` because that function already has workload implied in its name.

Reply:
do we use the returned `[]string` anywhere...? or just the `error`

Reply:
Yes, the valid scaling fields are used as a replacement for `hasScalingFieldsSet()`. They're just returned as a slice in case in future uses would need to have information on specifically which ones are set.

Then, the point of changing this to use reflect was to consolidate the places where changes would need to be made when adding new fields. I feel like if we're too concerned about the potential dangers of the reflect package, then the initial implementation I made would be better suited than the drilling. The initial implementation just hard-coded the invalid fields. Once again it's not ideal, but in all, that solution is the most readable and simple. The `reflect` package however minimizes future changes, though it does add more complications and risk if the file were overhauled in some way.

Overall I think the trade-off between readability and reproducibility is worth it in this case, and am in favor of keeping the reflect package in this instance. I also think that one drawback to this solution was that it was harder to implement, but as I've already written the code for it, the initial implementation is no longer an issue.